libfreeipmi/driver/ipmi-inteldcmi-driver.c

```diff
@@ -444,6 +445,19 @@ ipmi_inteldcmi_ctx_io_init (ipmi_inteldcmi_ctx_t ctx)
       goto cleanup;
     }
 
+  flags = fcntl(ctx->device_fd, F_GETFD);
+  if (flags == -1)
+    {
```

Comment:
Hmm, interesting.  Each Linux variant I have says -1 on error in the manpages.  == -1 for the F_SETFD seems fine then.